packages/gatsby-theme-docs/src/components/beta-flag.js

```diff
@@ -39,7 +38,7 @@ const getStyles = props => {
 const BetaFlag = props => {
   if (props.href) {
     return (
-      <Link href={props.href} css={getStyles(props)}>
+      <Link href={props.href} noUnderline={true} css={getStyles(props)}>
```

Comment:
FYI: the prop is called `noUnderline` on purpose, because the default behavior should be with underline and you can decide to explicitly opt-out.